packages/engine/src/interaction/systems/EquippableSystem.ts

```diff
@@ -51,56 +60,52 @@ export default async function EquippableSystem(world: World): Promise<System> {
-            userId: Engine.userId,
-            networkId: networkComponet.networkId,
-            attachmentPoint: attachmentPoint,
-            equip: true
-          })
-        )
+      if (collider) {
+        console.log('Change type to kinematic')
+        let phsyxRigidbody = collider.body as PhysX.PxRigidBody
+        ;(phsyxRigidbody as any)._type = BodyType.KINEMATIC
+        phsyxRigidbody.setRigidBodyFlag(PhysX.PxRigidBodyFlag.eKINEMATIC, true)
       }
     }
 
     for (const entity of equippableQuery()) {
       const equipperComponent = getComponent(entity, EquipperComponent)
-      const equippedComponent = getComponent(equipperComponent.equippedEntity, EquippedComponent)
-      const attachmentPoint = equippedComponent.attachmentPoint
-      const equippableTransform = getComponent(equipperComponent.equippedEntity, TransformComponent)
-      const handTransform = getHandTransform(entity, getParity(attachmentPoint))
-      const { position, rotation } = handTransform
-      equippableTransform.position.copy(position)
-      equippableTransform.rotation.copy(rotation)
+      const equippedEntity = equipperComponent.equippedEntity
+      if (equippedEntity) {
+        const isOwnedByMe = getComponent(equippedEntity, NetworkObjectOwnedTag)
+        if (isOwnedByMe) {
+          const equippedComponent = getComponent(equipperComponent.equippedEntity, EquippedComponent)
+          const attachmentPoint = equippedComponent.attachmentPoint
+          const equippableTransform = getComponent(equipperComponent.equippedEntity, TransformComponent)
+          const handTransform = getHandTransform(entity, getParity(attachmentPoint))
+          const { position, rotation } = handTransform
+
+          const collider = getComponent(equippedEntity, ColliderComponent)
+          if (collider) {
+            teleportRigidbody(collider.body, position, rotation)
+          }
+
+          equippableTransform.position.copy(position)
+          equippableTransform.rotation.copy(rotation)
+        }
+      }
     }
 
     for (const entity of equippableQuery.exit()) {
       const equipperComponent = getComponent(entity, EquipperComponent, true)
       const equippedEntity = equipperComponent.equippedEntity
-      const equippedComponent = getComponent(equippedEntity, EquippedComponent)
-      const attachmentPoint = equippedComponent.attachmentPoint
 
-      // const equippedTransform = getComponent(equippedEntity, TransformComponent)
-      // const collider = getComponent(equippedEntity, ColliderComponent)
-      // if (collider) {
-      //   // collider.body.type = BodyType.DYNAMIC
-      //   teleportRigidbody(collider.body, equippedTransform.position, equippedTransform.rotation)
-      // }
+      const equippedTransform = getComponent(equippedEntity, TransformComponent)
+      const collider = getComponent(equippedEntity, ColliderComponent)
+      if (collider) {
+        let phsyxRigidbody = collider.body as PhysX.PxRigidBody
```

Comment:
lets create some `changeRigidbodyType(body: PhysX.PxRigidBody, type: BodyType)` in `Physics.ts`  to abstract this into unified calls

Reply:
Added.